src/ore/src/tracing.rs

```diff
@@ -202,3 +222,58 @@ where
         Ok(())
     }
 }
+
+/// Helpers for working with OpenTelemetry
+pub mod otel {
+    use opentelemetry::global;
+    use opentelemetry::propagation::{Extractor, Injector};
+    use std::collections::HashMap;
+    use tracing_opentelemetry::OpenTelemetrySpanExt;
+
+    /// An opaque struct to propagate otel context's,
+    /// with various helper methods.
+    #[derive(Debug, Clone)]
+    pub struct OtelContext {
```

Comment:
nit: I think consistency with types in `tracing_opentelemetry` wins over conciseness here and we should call this `OpenTelemetryContext`.